### PR TITLE
tests: expand backup flow device tests

### PR DESCRIPTION
**Pull request by ciny**
This PR implements #551 

- simple tests are in test_msg_backup_device
- expanded tests (skip_backup) are in test_reset_backup_<BACKUP_TYPE>
- tests for backup to fail after it was done during reset were added to test_msg_resetdevice_<BACKUP_TYPE>

### CODE REVIEW

File: tests/device_tests/test_reset_backup_bip39.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
+# This file is part of the Trezor project.
```
Comment (tsusanka) [inline tests/device_tests/test_reset_backup_bip39.py]: Let's call this file `test_reset_delayed_backup_bip39` and the other analogically?

File: tests/device_tests/test_msg_backup_device.py [hunk 2]
```diff
@@ -118,8 +124,214 @@ def input_flow():
...
+
+    expected_ms = shamir.combine_mnemonics(MNEMONIC_SLIP39_ADVANCED_20)
+    actual_ms = shamir.combine_mnemonics(
+        mnemonics[:3] + mnemonics[5:8] + mnemonics[10:13]
+    )
+    assert expected_ms == actual_ms
+
+
+# we only test this with bip39 because the code path is always the same
+@pytest.mark.skip_t1
```
Comment (tsusanka) [inline tests/device_tests/test_msg_backup_device.py]: Couldn't we run this on T1 as well?

File: tests/device_tests/test_msg_backup_device.py [hunk 3]
```diff
@@ -118,8 +124,214 @@ def input_flow():
...
+    assert client.features.no_backup is True
+    assert client.features.needs_backup is False
+
+    # backup attempt should fail because no_backup=True
+    with pytest.raises(TrezorFailure, match="ProcessError: Seed already backed up"):
+        device.backup(client)
+
+
+# we only test this with bip39 because the code path is always the same
+@pytest.mark.skip_t1
```
Comment (tsusanka) [inline tests/device_tests/test_msg_backup_device.py]: Couldn't we run this on T1 as well?

File: tests/device_tests/test_msg_backup_device.py [hunk 4]
```diff
@@ -118,8 +124,214 @@ def input_flow():
...
+    assert client.features.unfinished_backup is True
+    assert client.features.no_backup is False
+
+    # Second attempt at backup should fail
+    with pytest.raises(TrezorFailure, match="ProcessError: Seed already backed up"):
+        device.backup(client)
+
+
+# we only test this with bip39 because the code path is always the same
+@pytest.mark.skip_t1
```
Comment (tsusanka) [inline tests/device_tests/test_msg_backup_device.py]: Couldn't we run this on T1 as well?

File: tests/device_tests/test_reset_backup_bip39.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+from trezorlib import device, messages
+from trezorlib.messages import BackupType, ButtonRequestType as B
+
+from ..common import click_through, generate_entropy, read_and_confirm_mnemonic
+
+EXTERNAL_ENTROPY = b"zlutoucky kun upel divoke ody" * 2
+OS_URANDOM = mock.Mock(return_value=EXTERNAL_ENTROPY)
+
+
+@pytest.mark.skip_t1
```
Comment (tsusanka) [inline tests/device_tests/test_reset_backup_bip39.py]: Couldn't we run this on T1 as well?

File: tests/device_tests/test_msg_backup_device.py [hunk 6]
```diff
@@ -118,8 +124,214 @@ def input_flow():
...
+
+    # backup attempt should fail because no_backup=True
+    with pytest.raises(TrezorFailure, match="ProcessError: Seed already backed up"):
+        device.backup(client)
+
+
+# we only test this with bip39 because the code path is always the same
+@pytest.mark.skip_t1
+@pytest.mark.setup_client(uninitialized=True)
+def test_interupt_backup_fails(client):
```
Comment (matejcik) [inline tests/device_tests/test_msg_backup_device.py]: typo: `interrupt`

File: tests/device_tests/test_reset_backup_bip39.py (hunk 1 above)
Comment (matejcik) [reply tests/device_tests/test_reset_backup_bip39.py]: I also think we can parametrize these tests.

File: tests/device_tests/test_msg_backup_device.py (hunk 2 above)
Comment (ciny) [reply tests/device_tests/test_msg_backup_device.py]: done

File: tests/device_tests/test_msg_backup_device.py (hunk 4 above)
Comment (ciny) [reply tests/device_tests/test_msg_backup_device.py]: done

File: tests/device_tests/test_msg_backup_device.py (hunk 3 above)
Comment (ciny) [reply tests/device_tests/test_msg_backup_device.py]: done

File: tests/device_tests/test_msg_backup_device.py (hunk 6 above)
Comment (ciny) [reply tests/device_tests/test_msg_backup_device.py]: fixed

File: tests/device_tests/test_reset_backup_bip39.py (hunk 5 above)
Comment (ciny) [reply tests/device_tests/test_reset_backup_bip39.py]: Here's a bit more complex. T1 and TT send different ButtonRequests during reset so we'd need separate input flow and expected responses for T1 and TT. I can have a look at it if it's something we want.

Comment (tsusanka) [reply tests/device_tests/test_reset_backup_bip39.py]: I think let's move on for now.

**Comment by tsusanka** [comment]
The monero tests was failing because an invalid Docker image on CI. This should be fixed now. Could you try out to remove the commit and force push?

**Comment by ciny** [comment]
> The monero test was failing because we had an invalid Docker image on CI. This should be fixed now. Could you try out to remove the commit and force push?

I removed it and it works.

**Comment by tsusanka** [comment]
@matejcik could you ACK and let's merge this?

**Pull request merged**